rotkehlchen/tests/unit/decoders/test_thegraph.py

```diff
@@ -0,0 +1,175 @@
+            event_subtype=HistoryEventSubType.DEPOSIT_ASSET,
+            asset=A_GRT,
+            balance=Balance(amount=FVal('998.98')),
+            location_label=ADDY_USER,
+            notes='Delegated 998.98 GRT to indexer 0x6125eA331851367716beE301ECDe7F38A7E429e7',
+            counterparty=CPT_THEGRAPH,
+            address=ADDY_THEGRAPH,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=358,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.APPROVE,
+            asset=A_GRT,
+            balance=Balance(amount=FVal('1.157920892373161954235709850E+59')),
+            location_label=ADDY_USER,
+            notes=(
+                f'Set GRT spending approval of {ADDY_USER} by {ADDY_THEGRAPH}'
+                f' to 115792089237316195423570985000000000000000000000000000000000'
+            ),
+            counterparty=None,
+            address=ADDY_THEGRAPH,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=360,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_GRT,
+            balance=Balance(amount=FVal('5.02')),
+            location_label=ADDY_USER,
+            notes='5.02 GRT has been burnt as delegation tax',
+            counterparty=CPT_THEGRAPH,
+            address=ADDY_THEGRAPH,
+        ),
+    ]
+    assert expected_events == events
+
+
+@pytest.mark.vcr()
+@pytest.mark.parametrize('ethereum_accounts', [[ADDY_USER]])
+def test_thegraph_undelegate(database, ethereum_inquirer):
+    tx_hash = deserialize_evm_tx_hash('0x5ca5244868d9c0d8c30a1cad0feaf137bd28acd9c3f669a09a3a199fd75ad25a')  # noqa: E501
+    events, _ = get_decoded_events_of_transaction(
+        evm_inquirer=ethereum_inquirer,
+        database=database,
+        tx_hash=tx_hash,
+    )
+    timestamp = TimestampMS(1691771855000)
+    expected_events = [
+        EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=0,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.SPEND,
+            event_subtype=HistoryEventSubType.FEE,
+            asset=A_ETH,
+            balance=Balance(amount=FVal('0.00307607001551556')),
+            location_label=ADDY_USER,
+            notes='Burned 0.00307607001551556 ETH for gas',
+            counterparty=CPT_GAS,
+        ), EvmEvent(
+            tx_hash=tx_hash,
+            sequence_index=297,
+            timestamp=timestamp,
+            location=Location.ETHEREUM,
+            event_type=HistoryEventType.INFORMATIONAL,
+            event_subtype=HistoryEventSubType.NONE,
+            asset=A_GRT,
+            balance=Balance(),
+            location_label=ADDY_USER,
+            notes=(
```

Comment:
try to keep in one line if possible or 2 if not. It's mostly about readability and splitting into multiple lines is hard to follow for big string.